core/src/banking_stage/scheduler_messages.rs

```diff
@@ -0,0 +1,76 @@
+use {
+    crate::immutable_deserialized_packet::ImmutableDeserializedPacket,
+    solana_sdk::transaction::SanitizedTransaction, std::sync::Arc,
+};
+
+/// A unique identifier for a transaction batch.
+/// The lower 64 bits are based on the order the transaction batch was scheduled.
+/// The upper 64 bits are an optional priority field - currently unused.
```

Comment:
Planning to experiment with using a skiplist to send messages, so that higher-priority batches can skip to the front.

Included the extra bits here already, but re-considering if I should do it here or only once I see it has value.

Reply:
Just another possibility, that `priority field` does have to be bundled with `unique id`, since they are two separate thing. This applies to `TransactionId` below as well. Perhaps can define `ConsumeWork` as 
```
ConsumeWork {
  transaction_batch_unique_id: u64,
  transaction_batch_priority: u64,  // unused
  transactions: &[SanitizedTransaction], 
  transaction_unique_ids: &[TransactionId],  // must be same length as transactions'
  transaction_priority: &[u64], // must be same length as transactions'
}
```

Reply:
If convinced to bundle unique_id and priority, then a struct might be more readable/changeable instead of bit encoding.

Reply:
Not 100% married to them being bundled, but I can explain a few reasons I chose to do so (after having them originally being separate).

1. Faster comparisons ([quick test example in godbolt](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:rust,selection:(endColumn:24,endLineNumber:10,positionColumn:24,positionLineNumber:10,selectionStartColumn:24,selectionStartLineNumber:10,startColumn:24,startLineNumber:10),source:'%23%5Bderive(Ord,+PartialOrd,+PartialEq,+Eq)%5D%0Apub+struct+BitField(u128)%3B%0A%0A%23%5Bderive(PartialOrd,+PartialEq,+Eq)%5D%0Apub+struct+SepField+%7B%0A++++a:+u64,%0A++++b:+u64,%0A%7D%0A%0Aimpl+Ord+for+SepField+%7B%0A++++fn+cmp(%26self,+other:+%26Self)+-%3E+std::cmp::Ordering+%7B%0A++++++++match+self.a.cmp(%26other.a)+%7B%0A++++++++++++std::cmp::Ordering::Equal+%3D%3E+%7B%0A++++++++++++++++self.b.cmp(%26other.b)%0A++++++++++++%7D,%0A++++++++++++ordering+%3D%3E+ordering%0A++++++++%7D%0A++++%7D%0A%7D%0A%0A//+impl+PartialOrd+for+SepField+%7B%0A//+++++fn+partial_ord(%26self,+other:+%26self)+-%3E+%0A//+%7D%0A%0A%0Apub+fn+check_bit(lhs:+%26BitField,+rhs:+%26BitField)+-%3E+bool+%7B%0A++++lhs+%3E+rhs%0A%7D%0A%0Apub+fn+check_sep(lhs:+%26SepField,+rhs:+%26SepField)+-%3E+bool+%7B%0A++++lhs+%3E+rhs%0A%7D%0A%0A'),l:'5',n:'0',o:'Rust+source+%231',t:'0')),k:50.05937206618502,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:r1650,deviceViewOpen:'1',filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:rust,libs:!(),options:'-C+opt-level%3D3',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+rustc+1.65.0+(Editor+%231)',t:'0')),k:49.940627933814994,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4))
2. Haven't tested it, but I'd expect by bundling them and using it for the priority queue sorting we'd see better perf, since there will be no conflicts (also has benefit of prioritizing transactions that arrived first if there is a tie in priority).

Reply:
"Faster comparisons" for sure, but at cost of less flexible to adapt in future.

ig my thought is `id` and `priority` ain't same kind of things that can be bundled together naturally. By that, I mean transactions are only compared/sorted by priority, or maybe something else down the road (say memory footprint, or execution cost); but not by its `unique id` (a bit of randomness in transactions are desired, i'd think). As for batch, there is no concept of priority for it yet. 

Then again, not strong on this one either. Perhaps can optimize for performance in later stage?